src/library/Connect/KubernetesRemoteEnvironmentManager.cs

```diff
@@ -114,22 +114,22 @@ internal class KubernetesRemoteEnvironmentManager : IRemoteEnvironmentManager
                         break;
                     // Clone pod spec and replace container with agent
                     case RemoteAgentHostingMode.NewPodWithContext:
-                        podAndContainer = await this._ClonePodAsync(remoteContainerConnectionDetails, cancellationToken);
+                        podAndContainer = await this._ClonePodAsync(remoteContainerConnectionDetails, localProcessConfig, cancellationToken);

```

Comment:
This code came to us with minimal UTs. So as we add new features / fix issues, we are trying to add UTs with the change that covers the parts being changed. It would be great if you could add one test file for this file in library.tests and add UTs to cover the 3 switches on this method.

Reply:
Yeah I was going to add Tests to this but I need advice on how to proceed here.

Todays code needs to be refactored if we want to add some meningfull tests for this function.
It calls alot of private functions and I havn't figured out how to test this in the current mock framework.
 
Do you want me to refactor it in this PR or create a seperate PR for the refactoring and adding tests?

Reply:
@Eneuman, I was hoping that mocking the variables used for the switch statements was enough. If this is not enough, then yes we can leave it to a later refactor PR.